client/v2/client.go

```
@@ -0,0 +1,371 @@
+
+	// Write consistency is the number of servers required to confirm write
+	WriteConsistency string
+}
+
+type Client interface {
+	// Write takes a BatchPoints object and writes all Points to InfluxDB.
+	Write(bp BatchPoints) error
+
+	// Query makes an InfluxDB Query on the database
+	Query(q Query) (*Response, error)
+}
+
+// NewClient creates a client interface from the given config.
+func NewClient(conf Config) Client {
+	if conf.UserAgent == "" {
+		conf.UserAgent = "InfluxDBClient"
+	}
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: conf.InsecureSkipVerify,
+		},
+	}
+	return &client{
+		url:       conf.URL,
+		username:  conf.Username,
+		password:  conf.Password,
+		useragent: conf.UserAgent,
+		httpClient: &http.Client{
+			Timeout:   conf.Timeout,
+			Transport: tr,
+		},
+	}
+}
+
+type client struct {
+	url        *url.URL
+	username   string
+	password   string
+	useragent  string
+	httpClient *http.Client
+}
+
+// BatchPoints is an interface into a batched grouping of points to write into
+// InfluxDB together.
+type BatchPoints interface {
+	// AddPoint adds the given point to the Batch of points
+	AddPoint(p *Point)
+	// Points lists the points in the Batch
+	Points() []*Point
+
+	// Precision returns the currently set precision of this Batch
+	Precision() string
+	// SetPrecision sets the precision of this batch.
+	SetPrecision(s string)
+
+	// Database returns the currently set database of this Batch
+	Database() string
+	// SetDatabase sets the database of this Batch
+	SetDatabase(s string)
+
+	// WriteConsistency returns the currently set write consistency of this Batch
+	WriteConsistency() string
+	// SetWriteConsistency sets the write consistency of this Batch
+	SetWriteConsistency(s string)
+
+	// RetentionPolicy returns the currently set retention policy of this Batch
+	RetentionPolicy() string
+	// SetRetentionPolicy sets the retention policy of this Batch
+	SetRetentionPolicy(s string)
```

Ditto for all these functions, I think they may need to return an error.